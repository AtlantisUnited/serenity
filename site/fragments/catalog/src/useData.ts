import { useQuery } from '@apollo/client'

import gql          from 'graphql-tag'

export const query = gql`
  query Categorys {
    categoryGroups {
      rows {
        id
        name
        children {
          id
          name
        }
      }
    }
  }
`

export const useData = () => {
  const { data } = useQuery(query)

  return data && data.categoryGroups ? data.categoryGroups.rows : []
}
