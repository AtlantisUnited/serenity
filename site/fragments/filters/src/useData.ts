import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client' // eslint-disable-line import/no-extraneous-dependencies

export const query = gql`
  query Categorys {
    categoryGroups {
      rows {
        id
        name
        children {
          id
          name
        }
      }
    }
  }
`

export const useData = () => {
  const { data } = useQuery(query)

  return data && data.categoryGroups ? data.categoryGroups.rows : []
}
