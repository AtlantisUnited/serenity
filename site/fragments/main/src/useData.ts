import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query Categories {
    categories {
      rows {
        id
        name
      }
    }
  }
`

export const useData = () => {
  const { data } = useQuery(query)

  return data && data.categories ? data.categories.rows : []
}
