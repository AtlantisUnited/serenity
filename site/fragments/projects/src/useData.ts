import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query SearchProjects($filters: SearchProjectsFilter, $query: String!) {
    search {
      projects(filters: $filters, query: $query) {
        rows {
          id
          name
          category {
            id
            name
          }
          description
          budget
          legalEntitiesOnly
          views
          replyCount
          publicationDate
        }
      }
    }
  }
`

export const useData = (searchQuery, searchCategory) => {
  const { data = {} }: any = useQuery(query, {
    variables: {
      query: searchQuery,
      filters: searchCategory
        ? {
            categoryId: searchCategory,
          }
        : null,
    },
  })

  return (data && data.search && data.search.projects && data.search.projects.rows) || []
}
