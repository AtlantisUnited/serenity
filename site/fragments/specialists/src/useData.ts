import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query SearchSpecialists($filters: SearchSpecialistsFilter, $query: String!) {
    search {
      specialists(filters: $filters, query: $query) {
        rows {
          id
          specialisation {
            main {
              name
            }
            additional {
              name
            }
          }
          description
          interaction {
            formOfWork
            name
          }
          profile {
            photo {
              url
            }
            personalInformation {
              firstName
              lastName
            }
          }
          portfolio {
            name
            images {
              id
              url
            }
          }
          rating
        }
      }
    }
  }
`

export const useData = (searchQuery, searchCategory) => {
  const { data = {} }: any = useQuery(query, {
    variables: {
      query: searchQuery,
      filters: searchCategory
        ? {
            specialisationId: searchCategory,
          }
        : null,
    },
  })

  return (data && data.search && data.search.specialists && data.search.specialists.rows) || []
}
