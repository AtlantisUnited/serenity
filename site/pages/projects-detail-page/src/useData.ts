import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const queryProjects = gql`
  query Projects($filters: ProjectsFilter) {
    projects(filters: $filters) {
      rows {
        name
        status
        category {
          name
        }
        description
        address {
          formatted
        }
        beginningOfWork
        photos {
          id
          url
        }
        publicationDate
        budget
        worksheet
        legalEntitiesOnly
        views
        replyCount
        owner {
          member {
            openProjects
            completedProjects
          }
          activity {
            last
          }
          profile {
            personalInformation {
              firstName
              lastName
            }
            photo {
              url
            }
            address {
              formatted
            }
          }
        }
      }
    }
  }
`

export const queryReplies = gql`
  query Me($projectId: ID) {
    me {
      profile {
        type
        photo {
          url
        }
      }
      replies(projectId: $projectId) {
        id
        status
        discussion {
          id
          messages {
            id
            content
            author {
              photo {
                url
              }
              personalInformation {
                firstName
                lastName
              }
            }
            member {
              id
              rating
              interaction {
                formOfWork
                name
              }
              specialisation {
                main {
                  name
                }
                additional {
                  name
                }
              }
            }
          }
        }
      }
    }
  }
`

export const useDataProjects = (id) => {
  const { data } = useQuery(queryProjects, {
    variables: {
      filters: {
        id,
      },
    },
  })

  return (data && data.projects && data.projects.rows[0]) || {}
}

export const useDataReplies = (id) => {
  const { data } = useQuery(queryReplies, {
    variables: {
      projectId: id,
    },
  })

  return (data && data.me && { profile: data.me.profile, replies: data.me.replies }) || {}
}
