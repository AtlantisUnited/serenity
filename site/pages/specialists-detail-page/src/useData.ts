import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query Specialists($filters: SpecialistsFilter) {
    specialists(filters: $filters) {
      rows {
        id
        reviewCount
        completedProjects
        specialisation {
          main {
            id
            name
          }
          additional {
            id
            name
          }
        }
        description
        interaction {
          numberOfEmployees
          formOfWork
          name
        }
        profile {
          photo {
            url
          }
          personalInformation {
            firstName
            lastName
          }
          contactInformation {
            phone {
              number
            }
          }
          address {
            formatted
          }
          website
        }
        portfolio {
          name
          images {
            id
            url
          }
        }
        rating
        reviews {
          id
          rating
          comment
          customer {
            personalInformation {
              firstName
              lastName
            }
            photo {
              url
            }
          }
        }
      }
    }
  }
`

export const useData = (id) => {
  const { data } = useQuery(query, {
    variables: {
      filters: {
        id,
      },
    },
  })

  return (data && data.specialists && data.specialists.rows[0]) || {}
}
