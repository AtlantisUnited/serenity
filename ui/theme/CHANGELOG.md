# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.5 (2021-05-10)

**Note:** Version bump only for package @ui/theme

## 0.0.4 (2020-09-28)

### Bug Fixes

- **accounts:** add emotion-server, delete font ([08acf5a](https://github.com/Atlantis-Lab/serenity/commit/08acf5a7169352a606925fdfbbaa18f9917a8106))

## 0.0.3 (2020-06-30)

**Note:** Version bump only for package @ui/theme

## 0.0.2 (2020-06-10)

**Note:** Version bump only for package @ui/theme
