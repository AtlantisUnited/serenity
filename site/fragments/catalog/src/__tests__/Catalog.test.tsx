import React              from 'react'

import { MockedProvider } from '@apollo/react-testing'
import { ThemeProvider }  from '@ui/theme'
import { render }         from '@testing-library/react'

import Catalog            from '..'
import { query }          from '../useData'

const mocks = [
  {
    request: { query },
    result: {
      data: {
        categoryGroups: {
          rows: [
            {
              id: 1,
              name: 'name',
              children: [
                {
                  id: 1,
                  name: 'name',
                },
              ],
            },
          ],
        },
      },
    },
  },
]

describe('Catalog fragment', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider>
            <Catalog />
          </ThemeProvider>
        </MockedProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
