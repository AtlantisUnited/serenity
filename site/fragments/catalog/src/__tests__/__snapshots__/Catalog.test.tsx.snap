// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog fragment snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="emotion-0"
    display="flex"
    height="100%"
  />
</DocumentFragment>
`;
